wire-ios-data-model/Tests/Source/ManagedObjectContext/DatabaseMigrationTests+UserClientUniqueness.swift

```diff
@@ -29,34 +29,114 @@ final class DatabaseMigrationTests_UserClientUniqueness: DatabaseBaseTest {
     private let tmpStoreURL = URL(fileURLWithPath: "\(NSTemporaryDirectory())databasetest/")
     private let dataModelName = "zmessaging"
 
+
+    func testThatItPerformsMigrationFromOldVersionsBefore107_ToCurrentModelVersion() throws {
+        // With version 107 and later we can not insert duplicated keys anymore!
+
+        let versions = [(84...96), (98...106)].joined().map {
+            "2.\($0).0"
+        }
+
+        try versions.forEach { initialVersion in
+            try migrateStoreToCurrentVersion(
+                sourceVersion: initialVersion,
+                preMigrationAction: insertDuplicates,
+                postMigrationAction: assertDuplicatesResolved
+            )
+            // clean after each test
+            self.clearStorageFolder()
+        }
+    }
+
+    func testThatItPerformsMigrationFromOlderVersion_2_105_0_ToCurrentModelVersion() throws {
+        let initialVersion = "2.101.0"
+        // 2.101.0 -> Wire v3.109.1
+
+        /*
+         Failed stores aka incompatible: 2.108, 2.105
+         Stores compatible: 2.109, 2.107
+         */
+        try migrateStoreToCurrentVersion(
+            sourceVersion: initialVersion,
+            preMigrationAction: insertDuplicates,
+            postMigrationAction: assertDuplicatesResolved
+        )
+    }
```

Comment:
Isn't the version already covered with test `testThatItPerformsMigrationFromOldVersionsBefore107_ToCurrentModelVersion()`?